plugins/pulp_rpm/plugins/importers/yum/associate.py

```diff
@@ -359,7 +359,8 @@ def _associate_unit(dest_repo, unit, config):
         models.PackageGroup,
         models.PackageCategory,
         models.PackageEnvironment,
-        models.PackageLangpacks
+        models.PackageLangpacks,
+        models.ModulemdDefaults
```

Comment:
copy of defaults should mimic same logic as yum_repo_metadata_file i.e. when copy occurs a new storage path should be calculated for copied modulemd-defaults.
if 2 module-defaults would share same storage_path in different repos, and it would be removed from 1 repo, then orhan cleaned up, second module-defaults would point to non-existent location

```
$ curl -i -H "Accept: application/json" -X POST -k -u admin:admin -d '{"source_repo_id": "fc28", "criteria": {"type_ids": ["modulemd_defaults"], "filters": {"unit": {}}}}' 'https://localhost/pulp/api/v2/repositories/copy-test/actions/associate/'
HTTP/1.1 202 ACCEPTED
$ pulp-admin rpm repo list
Warning: path should have mode 0700 because it may contain sensitive information: /home/ipanova/.pulp/

+----------------------------------------------------------------------+
                            RPM Repositories
+----------------------------------------------------------------------+

Id:                  fc28
Display Name:        None
Description:         None
Content Unit Counts: 
  Distribution:      1
  Modulemd:          5
  Modulemd Defaults: 3
  Package Langpacks: 1
  Rpm:               23

Id:                  copy-test
Display Name:        None
Description:         None
Content Unit Counts: 
  Modulemd Defaults: 3

> use pulp_database
switched to db pulp_database
> db.units_modulemd_defaults.find({'repo_id':'copy-test'})
{ "_id" : "5764333e-6e96-4d5a-a19f-0f2cf51c2c72", "pulp_user_metadata" : {  }, "_last_updated" : 1532617213, "_storage_path" : "/var/lib/pulp/content/units/modulemd_defaults/de/057d43335a94d3eea3ba2ecb50849fa6eccc43e9c79acc786bdfdb37914d75", "downloaded" : true, "name" : "nodejs", "repo_id" : "copy-test", "profiles" : "J\u0000\u0000\u0000\u00049\u0000\u0014\u0000\u0000\u0000\u00020\u0000\b\u0000\u0000\u0000default\u0000\u0000\u00048\u0000\u0014\u0000\u0000\u0000\u00020\u0000\b\u0000\u0000\u0000default\u0000\u0000\u00046\u0000\u0014\u0000\u0000\u0000\u00020\u0000\b\u0000\u0000\u0000default\u0000\u0000\u0000", "checksum" : "1e21417901e00aaab3008a0aea20775671dae9242ca16cf3b3d5896e2eba999f", "_ns" : "units_modulemd_defaults", "_content_type_id" : "modulemd_defaults" }
{ "_id" : "a0a46904-fdb2-4dde-8be6-4403d8a1d162", "pulp_user_metadata" : {  }, "_last_updated" : 1532617213, "_storage_path" : "/var/lib/pulp/content/units/modulemd_defaults/31/26b03db1a7acda68e240ad5b3a84b53625ebdff82110568eeb8cfd4cc5918d", "downloaded" : true, "name" : "reviewboard", "repo_id" : "copy-test", "stream" : "2.5", "profiles" : "\u001e\u0000\u0000\u0000\u00042.5\u0000\u0014\u0000\u0000\u0000\u00020\u0000\b\u0000\u0000\u0000default\u0000\u0000\u0000", "checksum" : "5659bef68645271341b1f38e3ffe8e93e5b6122aadbf189406cc17cacdb791dd", "_ns" : "units_modulemd_defaults", "_content_type_id" : "modulemd_defaults" }
{ "_id" : "aeb00d7f-67ad-4b68-960f-91b5a138b12e", "pulp_user_metadata" : {  }, "_last_updated" : 1532617213, "_storage_path" : "/var/lib/pulp/content/units/modulemd_defaults/fd/fe5817d520c3796edf956b5a43b5c0fb91392fb1cbad9e5be1e7e7b4744c96", "downloaded" : true, "name" : "django", "repo_id" : "copy-test", "profiles" : "\u001e\u0000\u0000\u0000\u00041.6\u0000\u0014\u0000\u0000\u0000\u00020\u0000\b\u0000\u0000\u0000default\u0000\u0000\u0000", "checksum" : "918320a560b67a815539ebf5863fada87c9b0a89f61bd35a5b769108b76e2ed5", "_ns" : "units_modulemd_defaults", "_content_type_id" : "modulemd_defaults" }
> db.units_modulemd_defaults.find({'repo_id':'fc28'})
{ "_id" : "1fb173f9-a96f-4996-9e7d-79fa0254a2cf", "pulp_user_metadata" : {  }, "_last_updated" : 1532612500, "_storage_path" : "/var/lib/pulp/content/units/modulemd_defaults/de/057d43335a94d3eea3ba2ecb50849fa6eccc43e9c79acc786bdfdb37914d75", "downloaded" : true, "name" : "nodejs", "repo_id" : "fc28", "profiles" : "J\u0000\u0000\u0000\u00049\u0000\u0014\u0000\u0000\u0000\u00020\u0000\b\u0000\u0000\u0000default\u0000\u0000\u00048\u0000\u0014\u0000\u0000\u0000\u00020\u0000\b\u0000\u0000\u0000default\u0000\u0000\u00046\u0000\u0014\u0000\u0000\u0000\u00020\u0000\b\u0000\u0000\u0000default\u0000\u0000\u0000", "checksum" : "1e21417901e00aaab3008a0aea20775671dae9242ca16cf3b3d5896e2eba999f", "_ns" : "units_modulemd_defaults", "_content_type_id" : "modulemd_defaults" }
{ "_id" : "58c04db5-94d9-4986-a45d-73c9da59c041", "pulp_user_metadata" : {  }, "_last_updated" : 1532612500, "_storage_path" : "/var/lib/pulp/content/units/modulemd_defaults/31/26b03db1a7acda68e240ad5b3a84b53625ebdff82110568eeb8cfd4cc5918d", "downloaded" : true, "name" : "reviewboard", "repo_id" : "fc28", "stream" : "2.5", "profiles" : "\u001e\u0000\u0000\u0000\u00042.5\u0000\u0014\u0000\u0000\u0000\u00020\u0000\b\u0000\u0000\u0000default\u0000\u0000\u0000", "checksum" : "5659bef68645271341b1f38e3ffe8e93e5b6122aadbf189406cc17cacdb791dd", "_ns" : "units_modulemd_defaults", "_content_type_id" : "modulemd_defaults" }
{ "_id" : "e46b6d4d-bd3f-4117-9a96-077ebdafa037", "pulp_user_metadata" : {  }, "_last_updated" : 1532612500, "_storage_path" : "/var/lib/pulp/content/units/modulemd_defaults/fd/fe5817d520c3796edf956b5a43b5c0fb91392fb1cbad9e5be1e7e7b4744c96", "downloaded" : true, "name" : "django", "repo_id" : "fc28", "profiles" : "\u001e\u0000\u0000\u0000\u00041.6\u0000\u0014\u0000\u0000\u0000\u00020\u0000\b\u0000\u0000\u0000default\u0000\u0000\u0000", "checksum" : "918320a560b67a815539ebf5863fada87c9b0a89f61bd35a5b769108b76e2ed5", "_ns" : "units_modulemd_defaults", "_content_type_id" : "modulemd_defaults" }

ll /var/lib/pulp/content/units/modulemd_defaults/de/057d43335a94d3eea3ba2ecb50849fa6eccc43e9c79acc786bdfdb37914d75
-rw-r--r--. 1 apache apache 147 Jul 26 15:41 /var/lib/pulp/content/units/modulemd_defaults/de/057d43335a94d3eea3ba2ecb50849fa6eccc43e9c79acc786bdfdb37914d75

$ curl -i -H "Accept: application/json" -X POST -k -u admin:admin -d '{"source_repo_id": "fc28", "criteria": {"type_ids": ["modulemd_defaults"], "filters": {"unit": {}}}}' 'https://localhost/pulp/api/v2/repositories/copy-test/actions/unassociate/'
HTTP/1.1 202 ACCEPTED

$ pulp-admin orphan remove --all

$ ll /var/lib/pulp/content/units/modulemd_defaults/de/057d43335a94d3eea3ba2ecb50849fa6eccc43e9c79acc786bdfdb37914d75
ls: cannot access '/var/lib/pulp/content/units/modulemd_defaults/de/057d43335a94d3eea3ba2ecb50849fa6eccc43e9c79acc786bdfdb37914d75': No such file or directory

```

Reply:
+1